src/bundle.c

```diff
@@ -29,15 +30,34 @@ static gboolean mksquashfs(const gchar *bundlename, const gchar *contentdir, GEr
 		goto out;
 	}
 
-	sproc = r_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_SILENCE,
-			&ierror, "mksquashfs",
-			contentdir,
-			bundlename,
-			"-all-root",
-			"-noappend",
-			"-no-progress",
-			"-no-xattrs",
-			NULL);
+	g_ptr_array_add(args, g_strdup("mksquashfs"));
+	g_ptr_array_add(args, g_strdup(contentdir));
+	g_ptr_array_add(args, g_strdup(bundlename));
+	g_ptr_array_add(args, g_strdup("-all-root"));
+	g_ptr_array_add(args, g_strdup("-noappend"));
+	g_ptr_array_add(args, g_strdup("-no-progress"));
+	g_ptr_array_add(args, g_strdup("-no-xattrs"));
+
+
+	if (r_context()->mksquashfs_args != NULL) {
+		gint mksquashfs_argcp;
+		g_auto(GStrv) mksquashfs_argvp = NULL;
+		res = g_shell_parse_argv(r_context()->mksquashfs_args, &mksquashfs_argcp, &mksquashfs_argvp, ierror);
+		if (!res) {
+			g_propagate_prefixed_error(
+					error,
+					&ierror,
```

Comment:
The error propagation functions expect a `GError*` for the inner error, not a `GError**` (see [`g_propagate_prefixed_error()`'s documentation](https://developer.gnome.org/glib/stable/glib-Error-Reporting.html#g-propagate-prefixed-error)), so please replace `&ierror` with `ierror`.

Reply:
I've fixed it to shorten the loop a little.